CentOS7でユニットテスト、E2Eテストを実行する

nanasess opened the pull request:
<!-- 以下を参考にコメントを作成してください。 -->

## 概要(Overview・Refs Issue)
<!-- PullRequestの目的、関連するIssue番号など -->
CentOS7 で PHPUnit、E2Eテスト を実行する。
以下データベースの下限環境を検証する目的
- mariadb 5.5.65(MySQL5.5互換)
- PostgreSQL 9.2.24

## 方針(Policy)
<!-- このPullRequestを作るにあたって考慮したものや除外した内容
  - 例）Symfony のXXにならって作成、3系で同様の仕様があったため など -->

Github Actions の `jobs.<job_id>.container` を使用して、 CentOS7 上の PHPUnit 及び Codeception を実行する

## 実装に関する補足(Appendix)
<!-- コードだけではわかりづらい点など、実装するにあたってレビューアに追加で伝えておきたいこと -->
- [ここのコメント](https://github.com/DDoSolitary/LxRunOffline/issues/88#issuecomment-525923925)を参考に、 systemctl を置き換えて コンテナ上の CentOS7 で systemctl を使用できるようにしている
- デフォルトでは特権ユーザーでJOBを実行するため、一般ユーザーを作成してテストする
- - `unknown error: DevToolsActivePort file doesn't exist` を防ぐため chromeOptions を追加

## テスト（Test)
<!-- テストを行っている範囲など、レビューアが安心できるような情報 -->
GitHub Actions が通るのを確認

## 相談（Discussion）
<!-- 相談したいことや意見を求めたいこと -->

~~以下のテストが落ちているがデータベースのバージョンに依存する箇所ではなさそうなので、とり急ぎは無視してもよいかも~~ 特権ユーザーで JOB を実行していたため、パーミッションの設定を無視されており、一般ユーザーで実行するようにしたところ解消しました。

 - `Eccube\Tests\Web\Admin\Content\CssControllerTest::test_routing_AdminContentCss_edit_failur`
- `Eccube\Tests\Web\Admin\Content\JsControllerTest::test_routing_AdminContentJs_edit_failure`

## マイナーバージョン互換性保持のための制限事項チェックリスト
<!-- マイナーバージョンでは、機能・プラグイン・デザインテンプレート互換性を損なう変更は原則取り込みません。 -->

- [x] 既存機能の仕様変更
- [x] フックポイントの呼び出しタイミングの変更
- [x] フックポイントのパラメータの削除・データ型の変更
- [x] twigファイルに渡しているパラメータの削除・データ型の変更
- [x] Serviceクラスの公開関数の、引数の削除・データ型の変更
- [x] 入出力ファイル(CSVなど)のフォーマット変更

## レビュワー確認項目

- [ ] 動作確認
- [ ] コードレビュー
- [ ] E2E/Unit テスト確認(テストの追加・変更が必要かどうか)
- [ ] 互換性が保持されているか
- [ ] セキュリティ上の問題がないか

nanasess committed:
Use centos7

nanasess committed:
Setup composer

nanasess committed:
CREATE DATABASE

nanasess committed:
PHPUnit

okazy commented:
貴重なCentOSでのテストありがとうございます。
test_routing_AdminContentCss_edit_failur が通ってからマージしたいと思います。

nanasess committed:
Merge remote-tracking branch 'upstream/4.0' into use-centos-container

nanasess committed:
Use composer v1

nanasess committed:
一般ユーザーでテストする

nanasess committed:
E2Eテストを追加

- `unknown error: DevToolsActivePort file doesn't exist` を防ぐため chromeOptions を追加

nanasess commented:
> test_routing_AdminContentCss_edit_failur が通ってから

テストを特権ユーザーで実行しているのが原因でした。一般ユーザーを作成してテストするよう修正しました

xuelian311 commented:
CentOS7のEOLが2024/6/30であと1年ちょっとですので、本件はマージせずクローズいたします。
